examples/allocation/tinygo/testdata/greet.go

```diff
@@ -72,13 +74,19 @@ func ptrToString(ptr uint32, size uint32) string {
 	}))
 }
 
-// stringToPtr returns a pointer and size pair for the given string in a way
-// compatible with WebAssembly numeric types.
+// stringToPtr returns a pointer and size pair to the underlying bytes of the given
+// string in a way compatible with WebAssembly numeric types. Since Go strings are
+// immutable, the pointer returned by stringToPtr must not be modified.
```

Comment:
I'd add another sentence here that ptr doesn't keep the string alive, just because I find it likely that this function (these 3 in fact) get copied around a lot.